homeassistant/components/generic/config_flow.py

```diff
@@ -57,7 +57,7 @@
     CONF_VERIFY_SSL: True,
 }
 
-SUPPORTED_IMAGE_TYPES = ["png", "jpeg", "svg+xml"]
+SUPPORTED_IMAGE_TYPES = ["png", "jpeg", "gif", "svg+xml"]
```

Comment:
```suggestion
SUPPORTED_IMAGE_TYPES = {"png", "jpeg", "gif", "svg+xml"}
```

Reply:
If you make this a set `fmt not in SUPPORTED_IMAGE_TYPES` doesn't have to do a linear search.  It doesn't matter much though with small lists.